mono/mini/interp/interp.c

```diff
@@ -5529,15 +5532,17 @@ interp_stop_single_stepping (void)
 	ss_enabled = FALSE;
 }
 
-int
-mono_ee_interp_version ()
+static int
+interp_ee_version (void)
 {
 	return MONO_EE_API_VERSION;
 }
 
 void
 mono_ee_interp_init (const char *opts)
 {
+	g_assert (mono_ee_api_version () == interp_ee_version ());
+
 	mono_native_tls_alloc (&thread_context_id, NULL);
```

Comment:
I think the registration function should just register the interpreter with the runtime, it shouldn't do too much initialization, because we don't know when the embedding app will call it, it can call it when nothing has been initialized yet.

Reply:
the embedding app has to call it before `mini_init ()`, that's for sure. I'm not sure what you mean by "too much initialization", it doesn't do a lot already.

Reply:
I mean it might call some init function which depends on the runtime being initialized.